### PR TITLE
Mailer modeling

### PR DESCRIPTION
**Pull request by msz-dev**
Mailer refactor, regarding the refactoring of the architecture on welcome email.

Now, the welcome email is send by the model, using the self (this) and life cycle hooks (with callbacks).

### CODE REVIEW

**Comment by msz-dev** [comment]
@gvillalta99 another one =D

File: app/models/author.rb [hunk 1]
```diff
@@ -1,3 +1,10 @@
 class Author < ApplicationRecord
   has_many :articles
+
+  after_create :welcome_mail
```
Comment (gvillalta99) [inline app/models/author.rb]: @msz-dev what is the difference between this callback and `after_commit :welcome_email on: :create`?

File: app/models/author.rb [hunk 2]
```diff
@@ -1,3 +1,10 @@
...
+  
+  def welcome_mail()
```
Comment (gvillalta99) [inline app/models/author.rb]: Those parenthesis are not needed ;)

File: app/workers/hard_worker.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class HardWorker
+  include Sidekiq::Worker
+
+  def perform(name, count)
+    # Do something
+  end
+end
```
Comment (gvillalta99) [inline app/workers/hard_worker.rb]: @msz-dev if this is not being used, please remove it

File: config/initializers/sidekiq.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+Sidekiq.configure_server do |config|
+  config.redis = { url: 'redis://localhost:6379/0', namespace: "app3_sidekiq_#{Rails.env}" }
+end
+
+Sidekiq.configure_client do |config|
+  config.redis = { url: 'redis://localhost:6379/0', namespace: "app3_sidekiq_#{Rails.env}" }
```
Comment (gvillalta99) [inline config/initializers/sidekiq.rb]: Next step, move sensitive information to an `.env` file

File: .vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+// Place your settings in this file to overwrite default and user settings.
```
Comment (gvillalta99) [inline .vscode/settings.json]: @msz-dev could add this file to `.gitignore` and remove it?

File: app/workers/hard_worker.rb (hunk 3 above)
Comment (msz-dev) [inline app/workers/hard_worker.rb]: @gvillalta99 Should i remove just the method or the whole class?

**Comment by msz-dev** [comment]
ok to test, @gvillalta99

**Pull request merged**